extensions/amp-story/1.0/live-story-manager.js

```diff
@@ -64,37 +69,28 @@ export class LiveStoryManager {
       'amp-story must contain id to use the live story functionality'
     );
 
+    Services.extensionsFor(this.ampdoc_.win).installExtensionForDoc(
+      this.ampdoc_,
+      'amp-live-list'
+    );
+
     this.storyEl_.insertBefore(liveListEl, this.storyEl_.firstElementChild);
   }
 
   /**
    * Updates the client amp-story with the changes from the server document.
-   *
-   * @param {?EventTarget} updatedStoryEl
-   * @param {!NodeList<!Element>} currentPages
    */
-  update(updatedStoryEl, currentPages) {
-    const newPageEls = devAssert(
-      updatedStoryEl,
-      'No updated story EventTarget was found.'
-    ).querySelectorAll('amp-story-page.amp-live-list-item-new');
-
-    let lastPageEl = currentPages[currentPages.length - 1];
+  update() {
+    const storyPages = this.storyEl_.querySelectorAll('amp-story-page');
+    const newPages = Array.prototype.filter.call(storyPages, page => {
+      return page.classList.contains('amp-live-list-item-new');
+    });
```

Comment:
Could you please use [lastChildElement](https://github.com/ampproject/amphtml/blob/master/src/dom.js#L372) instead? It'd be a bit faster as it starts with the last elements and stops on first match.